src/ODETools/ODEOperators.jl

```diff
@@ -8,13 +8,15 @@ struct Nonlinear <: OperatorType end
 struct Affine  <: OperatorType end
 struct Constant  <: OperatorType end
 
+
 """
 It represents the operator in an implicit ODE, i.e., A(t,u,∂tu) where the
 implicit PDE reads A(t,u,∂tu) = 0, when ∂tu is the time derivative of u.
 The trait `{C}` determines whether the operator is fully nonlinear, affine
 or constant in time.
 """
 abstract type ODEOperator{C<:OperatorType} <: GridapType end
+abstract type SecondOrderODEOperator{C} <: ODEOperator{C} end
```

Comment:
Just a minor thing: I would not extend `SecondOrderODEOperator` from `ODEOperator` since they have different interfaces. I.e., the code implemented for `ODEOperator` will not work if you pass a `SecondOrderODEOperator`.

Reply:
No need for another type in the new implementation